common/kvstore/src/main/java/org/apache/spark/util/kvstore/InMemoryStore.java

```diff
@@ -154,34 +201,21 @@ private InstanceList(Class<?> type) throws Exception {
       return ti.getAccessor(indexName);
     }
 
-    <T> int countingRemoveIf(Predicate<? super T> filter) {
-      Iterator<Map.Entry<Comparable<Object>, Object>> each = data.entrySet().iterator();
-      int count = 0;
+    @SuppressWarnings("unchecked")
+    int countingRemoveIfByKey(String index, Collection keys) {
+      KVTypeInfo.Accessor getter = ti.getAccessor(index);
+      Predicate<? super T> filter = getPredicate(getter, keys);
```

Comment:
consider not binding `getter` and just doing `getPredicate(ti.getAccessor(index), keys)`